bip-update-process.md

```diff
@@ -0,0 +1,646 @@
+they are in active use or not. It is not intended for BIPs to list additional implementations beyond the reference
+implementation: the BIPs repository is not a signpost where to find implementations.[^OtherImplementations] After a BIP
+is advanced to Proposed, it is up to the bitcoin community to evaluate, adopt, ignore, or reject a BIP. Individual
+bitcoin projects are encouraged to publish a list of BIPs they implement. A good example of this at the time of writing
+this BIP can be observed in [Bitcoin Core’s doc/bips.md
+file](https://github.com/bitcoin/bitcoin/blob/master/doc/bips.md).
+
+### Transferring BIP Ownership
+
+It occasionally becomes necessary to transfer ownership of BIPs to a new author. In general, we would like to retain
+the original author as a co-author of the transferred BIP, but that is up to the original author. A good reason to
+transfer ownership is because the original author no longer has the time or interest in updating it or following through
+with the BIP process, or has fallen off the face of the 'net (i.e. is unreachable or not responding to email). A bad
+reason to transfer ownership is because someone doesn't agree with the direction of the BIP. We try to build consensus
+around a BIP, but if that's not possible, rather than fighting over control, the dissenters should supply a competing
+BIP.
+
+If someone is interested in assuming ownership of a BIP, they should send an email asking to take over, addressed to the
+original author and the BIP editors. If the author does not respond to email in a timely manner, the BIP editors will
+make a unilateral decision (which may be amended should the original author make a delayed reappearance).
+
+## BIP Licensing
+
+### Specification
+
+Each new BIP must identify at least one acceptable license in its preamble. Licenses must be referenced per their
+respective [SPDX License identifier](https://spdx.org/licenses). New BIPs may be accepted with the licenses described
+below.
+
+For example, a preamble might include the following License header:
+
+    License: CC0-1.0
+             GNU-All-Permissive
+
+In this case, the BIP text is fully licensed under both the Creative Commons CC0 1.0 Universal license as well as the
+GNU All-Permissive License, and anyone may modify and redistribute the text provided they comply with the terms of
+*either* license. In other words, the license list is an "OR choice", not an "AND also" requirement.
+
+It is also possible to license source code differently from the BIP text. An optional License-Code header is placed
+after the License header. Again, each license must be referenced by their respective abbreviation given below.
+
+For example, a preamble specifying the optional License-Code header might look like:
+
+    License:         CC0-1.0
+    License-Code:    MIT
+
+In this case, the code in the BIP is not available under CC0-1.0, but is only available under the terms of the MIT
+License.
```

Comment:
I suggest adding the recommendation that any source code files (or source directories) should specify their license, as is common in software (e.g., with a license header or a LICENSE/COPYING file). This makes sense because implementations are often copied without the rest of the BIP.

I also suggest making any "test vectors" / known input tests available under CC0 or GNU All Permissive. Everyone wants to copy test vectors in their implementations, so having them under a license different than your code is very annoying.

I also suggest allowing for "License-Code" to be something like "MIT (except where stated otherwise)" for complex cases, e.g., when files are under different licenses. A good example is the test vector thing. Then at least we don't need to bother with the header.

Reply:
I'm not sure I understand. This would change a recommendation (and not a default), and only for future BIPs, no?

Reply:
> I also suggest allowing for "License-Code" to be something like "MIT (except where stated otherwise)" for complex cases, e.g., when files are under different licenses. A good example is the test vector thing. Then at least we don't need to bother with the header.

Oh, I think I read that as proposing it as a new policy for the whole repository rather than an option for a new BIP.

I have added:

>The licenses listed in the License-Code header apply to all source directories, source code files, and test vectors provided with the BIP except when the file header or a LICENSE file in a directory states otherwise.